Fix null error occurring when handling the Player Interact Event.

21.09 01:01:05 [Server] Server thread/ERROR Could not pass event PlayerInteractEvent to Fiefs v0.9
21.09 01:01:05 [Server] INFO org.bukkit.event.EventException: null
21.09 01:01:05 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:481) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.PlayerInteractManager.a(PlayerInteractManager.java:447) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:1356) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.PacketPlayInUseItem.a(PacketPlayInUseItem.java:27) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.PacketPlayInUseItem.a(PacketPlayInUseItem.java:1) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(SourceFile:144) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:05 [Server] INFO at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(SourceFile:118) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:07 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:942) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:07 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:935) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:07 [Server] INFO at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeAll(SourceFile:103) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:07 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.sleepForTick(MinecraftServer.java:918) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:07 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:851) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:07 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$0(MinecraftServer.java:164) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:07 [Server] INFO at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
21.09 01:01:07 [Server] INFO Caused by: java.lang.NullPointerException
21.09 01:01:07 [Server] INFO at dansplugins.fiefs.eventhandlers.InteractionHandler.shouldEventBeCancelled(InteractionHandler.java:199) ~[?:?]
21.09 01:01:07 [Server] INFO at dansplugins.fiefs.eventhandlers.InteractionHandler.handle(InteractionHandler.java:82) ~[?:?]
21.09 01:01:07 [Server] INFO at sun.reflect.GeneratedMethodAccessor260.invoke(Unknown Source) ~[?:?]
21.09 01:01:07 [Server] INFO at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
21.09 01:01:07 [Server] INFO at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
21.09 01:01:07 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.16.5.jar:2991-Spigot-018b9a0-f3f3094]
21.09 01:01:07 [Server] INFO ... 20 more

--- a/src/main/java/dansplugins/fiefs/eventhandlers/InteractionHandler.java
+++ b/src/main/java/dansplugins/fiefs/eventhandlers/InteractionHandler.java
@@ -38,7 +38,7 @@ public class InteractionHandler implements Listener {
         }
 
         if (shouldEventBeCancelled(claimedChunk, player)) {
-            if (Fiefs.getInstance().isDebugEnabled()) { System.out.println("[DEBUG] Cancelling Block Break event."); }
+            Logger.getInstance().log("Cancelling Block Break event.");
             event.setCancelled(true);
         }
     }
@@ -59,7 +59,7 @@ public class InteractionHandler implements Listener {
         }
 
         if (shouldEventBeCancelled(claimedChunk, player)) {
-            if (Fiefs.getInstance().isDebugEnabled()) { System.out.println("[DEBUG] Cancelling Block Place event."); }
+            Logger.getInstance().log("Cancelling Block Place event.");
             event.setCancelled(true);
         }
     }
@@ -80,7 +80,7 @@ public class InteractionHandler implements Listener {
         }
 
         if (shouldEventBeCancelled(claimedChunk, player)) {
-            if (Fiefs.getInstance().isDebugEnabled()) { System.out.println("[DEBUG] Cancelling Player Interact event."); }
+            Logger.getInstance().log("Cancelling Player Interact event.");
             event.setCancelled(true);
         }
     }
@@ -99,9 +99,7 @@ public class InteractionHandler implements Listener {
             location = armorStand.getLocation();
         }
         else if (clickedEntity instanceof ItemFrame) {
-            if (Fiefs.getInstance().isDebugEnabled()) {
-                System.out.println("DEBUG: ItemFrame interaction captured in PlayerInteractAtEntityEvent!");
-            }
+            Logger.getInstance().log("DEBUG: ItemFrame interaction captured in PlayerInteractAtEntityEvent!");
             ItemFrame itemFrame = (ItemFrame) clickedEntity;
 
             // get chunk that armor stand is in
@@ -138,7 +136,7 @@ public class InteractionHandler implements Listener {
 
     @EventHandler()
     public void handle(PlayerBucketFillEvent event) {
-        if (Fiefs.getInstance().isDebugEnabled()) { System.out.println("DEBUG: A player is attempting to fill a bucket!"); }
+        Logger.getInstance().log("A player is attempting to fill a bucket!");
 
         Player player = event.getPlayer();
 
@@ -190,11 +188,15 @@ public class InteractionHandler implements Listener {
 
     private boolean shouldEventBeCancelled(ClaimedChunk claimedChunk, Player player) {
         if (claimedChunk == null) {
-            if (Fiefs.getInstance().isDebugEnabled()) { System.out.println("[Fiefs] Claimed chunk was null."); }
+            Logger.getInstance().log("Claimed chunk was null.");
             return false;
         }
         Fief chunkHolder = PersistentData.getInstance().getFief(claimedChunk.getFief());
         Fief playersFief = PersistentData.getInstance().getFief(player);
+
+        if (playersFief == null) {
+            return true;
+        }
 
         boolean claimedLandProtected = (boolean) playersFief.getFlags().getFlag("claimedLandProtected");
 

--- a/src/main/java/dansplugins/fiefs/services/LocalChunkService.java
+++ b/src/main/java/dansplugins/fiefs/services/LocalChunkService.java
@@ -34,7 +34,7 @@ public class LocalChunkService {
 
     public boolean attemptToClaimChunk(Chunk chunk, Fief fief, Player player) {
         if (!MedievalFactionsIntegrator.getInstance().getAPI().isChunkClaimed(chunk)) {
-            player.sendMessage(ChatColor.RED + "You can only claim land which belongs to your faction.");
+            player.sendMessage(ChatColor.RED + "You can't claim land that your faction hasn't claimed.");
             return false;
         }
 

